client/src/actions/authActions.js

```diff
@@ -60,4 +60,26 @@ export const logoutUser = () => dispatch => {
 	dispatch(setCurrentUser({}));
 };
 
-// TODO: Add edit user
+// Edit profile
+export const editProfile = (user, profileData, history) => dispatch =>
+	axios
+		.put(`api/user/${user.id}`, profileData)
+		.then(res => {
+			// Edit current user profile
+			dispatch({
+				type: EDIT_PROFILE,
+				payload: res.data.user,
+			});
+		})
+		.then(() => history.push('/profile'))
+		.catch(err =>
+			dispatch({
+				type: GET_ERRORS,
+				payload: err.response.data,
+			})
+		);
+
+/*
+	The problem above is that if the user changes their profile and then refreshes the app, the old info from the JWT token will be loaded.
```

Comment:
Is there any way that we can refresh token here? Purge old token from localStorage and request a new one from the backend. 🤔 

If we don't have this functionality already maybe we can implement something like a token refresher on the backend.

Reply:
Good idea. I added a backend route that sets a new token.